OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -102,17 +102,20 @@ public void onReceivedError(WebView view, int errorCode,
                     @TargetApi(Build.VERSION_CODES.M)
                     public void onReceivedHttpError(WebView view, WebResourceRequest request,
                                                     WebResourceResponse errorResponse) {
-                        didReceiveError = true;
-                        switch (errorResponse.getStatusCode()) {
-                            case HttpStatus.FORBIDDEN:
-                            case HttpStatus.UNAUTHORIZED:
-                            case HttpStatus.NOT_FOUND:
-                                EdxCookieManager.getSharedInstance(getContext())
-                                        .tryToRefreshSessionCookie();
-                                break;
+                        // If error occurred for web page request
+                        if (request.getUrl().toString().equals(view.getUrl())) {
```

Comment:
This callback ([onReceivedHttpError](https://developer.android.com/reference/android/webkit/WebViewClient.html#onReceivedHttpError(android.webkit.WebView, android.webkit.WebResourceRequest, android.webkit.WebResourceResponse))) could be called for any resource (iframe, image, etc), not just for the main page.

While [onReceivedError](https://developer.android.com/reference/android/webkit/WebViewClient.html#onReceivedError(android.webkit.WebView, int, java.lang.String, java.lang.String)) callback will be called for errors that are unrecoverable and errorCode parameter corresponds to one of the ERROR_* constants that are defined on the top of page, in any of these error we should not load page.

Ref: https://developer.android.com/reference/android/webkit/WebViewClient.html#onReceivedError(android.webkit.WebView, int, java.lang.String, java.lang.String)